components/style/values/specified/basic_shape.rs

```diff
@@ -465,28 +450,18 @@ pub struct Ellipse {
     pub position: Position,
 }
 
-
 impl Ellipse {
     #[allow(missing_docs)]
     pub fn parse_function_arguments(context: &ParserContext, input: &mut Parser) -> Result<Ellipse, ()> {
-        let (a, b) = input.try(|input| -> Result<_, ()> {
-            Ok((try!(ShapeRadius::parse(context, input)), try!(ShapeRadius::parse(context, input))))
+        let (a, b) = input.try(|i| -> Result<_, ()> {
```

Comment:
Thought the same, but it looks like `rustc` cannot infer the type of error if I remove it. That's probably the reason why it exists in the first place.

    error[E0282]: unable to infer enough type information about `E`
       --> /media/Ubuntu/servo/components/style/values/specified/basic_shape.rs:456:28
        |
    456 |         let (a, b) = input.try(|i| {
        |                            ^^^ cannot infer type for `E`
        |
        = note: type annotations or generic parameter binding required

I could use `and_then`, but that might get a bit nasty.